### DISCUSSION TITLE
Which main branch to use, accounting for possible divergence w.r.t. original repo

### DISCUSSION THREAD

**Discussion by nunoguedelha** [ami-iit/openmct#4] [Q&A]
I was wondering if we should consider a single main branch where to integrate all the PRs, considering what is being discussed in https://github.com/ami-iit/openmct/discussions/3, and considering that main branch would probably diverge, at some point, from the branch originally tracked in https://github.com/nasa/openmct.

In the discussion that follows, I will refer to https://github.com/nasa/openmct as `nasa/openmct` repo, branches living in the forked repo as "fork branches", and branches living in `nasa/openmct` repo as "original branches".

`nasa/openmct` repo has a quite complex branching scheme (`main`, `master`, `<tag>-stable`, `<tag>-master`, etc), so many possible original branches we might want to hook-up to depending on our needs, we should choose a proper way to track which release we are based on while handling divergences. We list below a few choices among others:
1. Have a fork branch for each of the original branches, allowing the fork branches to diverge from the original ones. At all times we would easily know on which original branch it is based on.
2. Use a fork main branch with a unique name not used by the original repository, for instance `iit-main` or `iit-intgr-main`. this branch would be able to be reset to any commit, tag or branch from the original repository, or diverge, integrating new commits existing only in the forked repo.

I think the first solution is cleaner and easier for tracking any change of release (this choice might be obvious to others 😄 ). FYI, currently, the `openmct` version set in `yarp-openmct` dependencies is the tag 1.7.8 (https://github.com/ami-iit/yarp-openmct/blob/eb006ca09ceda1643411ad420f4a33ac52cd5d82/package.json#L14)
```
    "openmct": "https://github.com/nasa/openmct.git#1.7.8",
```
An original branch, `1.7.8`, coincides with the release tag `1.7.8`.

For instance, for integrating in our fork the fix to #1 , we could merge the PR in our forked branch `1.7.8`.

**Comment by nunoguedelha** [comment]
CC @traversaro , @riccardogrieco, @S-Dafarra

**Comment by nunoguedelha** [comment]
> An original branch, `1.7.8`, coincides with the release tag `1.7.8`.
> 
> For instance, for integrating in our fork the fix to #1 , we could merge the PR in our forked branch `1.7.8`.

Actually, https://github.com/nasa/openmct repository is using tags and branches with common names, which creates collisions and is a bad practice.

We could instead use a fork branch `1.7.8-iit` to start from an original tag `1.7.8` and integrate further fixes.

Or we just stick to the `master` which is the default. I don't know how stable it is, but for sure it has the more recent changes. By the way I have to check if issue #1 and the delays still occur in the `master` or the latest release `v2.0.3`.

**Comment by traversaro** [comment]
Probably `baseBranchName` + `-iit`  suffix is the best option as you suggest, for the branch from which to start we can try `master` and see if everything works as expected with that, otherwise try some older version.

**Reply by nunoguedelha** [reply]
So far I was using tag `1.7.8` which was the latest at the moment (October 2021). I haven't changed it since then.

Indeed I was planning to try on master in order to check the issues we encountered so far. But for selecting a stable version, I would go for the latest release `v2.0.3` (**2 May 2022**) if the functional status (from our point of view) is the same as mater. Do you agree?

**Reply by traversaro** [reply]
Sure, v2.0.3 seems fine as well.
